Mathlib/Data/Matrix/Invertible.lean

```diff
@@ -171,6 +172,42 @@ theorem invOf_add_mul_mul [Invertible (A + U*C*V)] :
   letI := invertibleAddMulMul A U C V
   convert (rfl : ⅟(A + U*C*V) = _)
 
+variable (u v : n → α)
+
+theorem add_col_mul_row_mul_invOf_eq_one
+    [Invertible (1 + row m v * ⅟A * col m u)] :
+    (⅟A - (⅟A * col m u * ⅟(1 + row m v * ⅟A * col m u) * row m v * ⅟A)) *
+      (A + col m u * row m v) = 1 := by
+  haveI : Invertible (1 : Matrix m m α) := invertibleOne
+  haveI : Invertible (⅟ 1 + row m v * ⅟A * col m u) := by simpa
+  simpa using add_mul_mul_invOf_mul_eq_one' A (col m u) 1 (row m v)
+
+/-- Like `Matrix.add_col_mul_row_mul_invOf_eq_one`, but with multiplication reversed. -/
+theorem add_col_mul_row_mul_invOf_eq_one'
+    [Invertible (1 + row m v * ⅟A * col m u)] :
+    (A + col m u * row m v) *
+      (⅟A - (⅟A * col m u * ⅟(1 + row m v * ⅟A * col m u) * row m v * ⅟A)) = 1 := by
+  haveI : Invertible (1 : Matrix m m α) := invertibleOne
+  haveI : Invertible (⅟ 1 + row m v * ⅟A * col m u) := by simpa
+  simpa using add_mul_mul_invOf_mul_eq_one A (col m u) 1 (row m v)
+
+variable (m) in
+/-- If matrix `A` and the scalar `(1 + row m v * ⅟A * col m u)` are invertible,
+then so is (A + col m u * row m v) -/
+def invertibleAddColMulRow [Invertible (1 + row m v * ⅟A * col m u)] :
+    Invertible (A + col m u * row m v) where
+  invOf := (⅟A - (⅟A * col m u * ⅟(1 + row m v * ⅟A * col m u) * row m v * ⅟A))
+  invOf_mul_self := add_col_mul_row_mul_invOf_eq_one _ _ _
+  mul_invOf_self := add_col_mul_row_mul_invOf_eq_one' _ _ _
+
+/-- The **Sherman Morrison Rank-1 Update** (`⅟` version). -/
+theorem invOf_add_col_mul_row_mul [Invertible (1 + row m v * ⅟A * col m u)]
+    [Invertible (A + col m u * row m v)] :
+    ⅟(A + col m u * row m v) =
+      (⅟A - (⅟A * col m u * ⅟(1 + row m v * ⅟A * col m u) * row m v * ⅟A)) := by
```

Comment:
Looking at this again, you can replace `col m u` with `B` and `row m v` with `D`, and the proof still goes through; I think you should make that change, but what you end up with is then what the matrix cookbook calls the "Kailath Variant".

Perhaps [my matrix cookbook formalization](https://github.com/eric-wieser/lean-matrix-cookbook/blob/5d8158c6d9a4f202f15d9f837997ca216e001603/MatrixCookbook/3Inverses.lean#L130-L150) is relevant here; to make it actually be about rows and columns, I state it as `1 + c ⬝ᵥ ⅟A *ᵥ b` (and use the Kailath result in the proof).